src/instructlab/model/serve.py

```diff
@@ -49,9 +50,9 @@
     "--backend",
     type=click.STRING,  # purposely not using click.Choice to allow for auto-detection
```

Comment:
You can use choice with an implicit `default=None` for auto-detection.

```python
import click


@click.command()
@click.option(
    "--backend",
    type=click.Choice(["llama-cpp", "vllm"]),
)
def test(backend):
    print(backend)


test()
```

```console
$ python3 test.py
None
$ python3 test.py --backend vllm
vllm
$ python3 test.py --help
Usage: test.py [OPTIONS]

Options:
  --backend [llama-cpp|vllm]
  --help                      Show this message and exit.

Reply:
Addressed, thanks for the suggestion.